fix fygar fire through moved rocks bug

pdPinho opened the pull request:
The current fygar's fire behaviour doesn't work correctly if he's facing a rock that has been moved.

This happens because this object's fire only checks for the initial rock positions.

With this fix, we're updating the rocks positions within their designated place every frame - which, even though not efficient, it fixes the given problem.

pdPinho committed:
fix fygar fire through moved rocks bug

dgomes commented on characters.py:
```diff
@@ -105,6 +105,10 @@ def to_dict(self):
     def __str__(self):
         return f"Rock({self.pos})"
 
+    @property
```
Esta property é genérica, muda para ficar na class mãe

dgomes commented on game.py:
```diff
@@ -294,6 +294,8 @@ async def next_frame(self):
         for rock in self._rocks:
             rock.move(self.map, digdug=self._digdug, rocks=self._rocks)
 
+        self.map.rocks_spawn = self._rocks # update rocks for mapa is_blocked
```
certo... isto actualiza

Mas a variável é rocks_**spawn**... e agora já não é...

dgomes commented on mapa.py:
```diff
@@ -134,6 +141,8 @@ def is_blocked(self, pos, traverse):
         x, y = pos
         if x not in range(self.hor_tiles) or y not in range(self.ver_tiles):
             return True
+        if pos in self._rocks:
```
Estas linha estiveram numa versão anterior e sairam ;)

dgomes reviewed (changes requested):
O problema está bem identificado, mas solução não deve ser ao nível do mapa pois estamos a falar de elementos (rocks) que não pertencem mais ao mapa.

Isto terá q ser resolvido ao nível do método disparo de fogo

pdPinho committed:
moved logic to fygar

pdPinho commented:
Moved the logic to the fygar object.

Removed redundancy - such as the position property (didn't notice that it's parent already owned such a property).

dgomes commented on characters.py:
```diff
@@ -358,8 +359,10 @@ def move(self, mapa, digdug, enemies, rocks):
-                ):  # Make sure we don't fire on ourselves:
+                    pos not in self.fire and
+                    pos != self.pos and
+                    pos not in rocks_pos
```
rock_pos só é usado aqui... coloca a list compreension directo

dgomes commented on game.py:
```diff
@@ -294,6 +294,7 @@ async def next_frame(self):
         for rock in self._rocks:
             rock.move(self.map, digdug=self._digdug, rocks=self._rocks)
 
+
```
```suggestion
```

pdPinho committed:
removed redundancy

dgomes commented:
Obrigado! 🎉
